examples/hybrid_compute/Quantinuum Hybrid Compute via pytket.ipynb

```diff
@@ -0,0 +1,887 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "68cd2200",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pathlib \n",
+    "\n",
+    "from pytket import wasm\n",
+    "\n",
+    "from pytket.circuit import (\n",
+    "    Circuit, BitRegister, Qubit, OpType,\n",
+    "    if_bit, if_not_bit,\n",
+    "    reg_eq, reg_geq, reg_gt, reg_leq, reg_lt, reg_neq\n",
+    ")\n",
+    "\n",
+    "from pytket.circuit.display import render_circuit_jupyter\n",
+    "from pytket.extensions.quantinuum import QuantinuumBackend"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "77c2719f",
+   "metadata": {},
+   "source": [
+    "#### Select Device\n",
+    "\n",
+    "Select device and login to the Quantinuum API using your credentials."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "c6f43b91",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Enter your Quantinuum email: megan.l.kohagen@quantinuum.com\n",
+      "Enter your Quantinuum password: ········\n"
+     ]
+    }
+   ],
+   "source": [
+    "machine = 'H1-1E'\n",
+    "backend = QuantinuumBackend(device_name=machine)\n",
+    "backend.login()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f534ff56",
+   "metadata": {},
+   "source": [
+    "#### Set up Circuit\n",
+    "\n",
+    "Set up the RUS circuit. This involves 2 steps:\n",
+    "1. Set up Wasm File handler. This checks the Wasm file and can list the available functions within the file. \n",
+    "2. Set up the pytket circuit, which calls the Wasm file handler."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "d9c12fa4",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Functions in wasm file with the uid f63152fe288eb6dc9e9cf58f7a02194f:\n",
+      "function 'init' with 0 i32 parameter(s) and 0 i32 return value(s)\n",
+      "function 'add_count' with 2 i32 parameter(s) and 1 i32 return value(s)\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "rus_dir = pathlib.Path().cwd().joinpath('repeat_until_success')\n",
+    "wasm_file = rus_dir.joinpath('rus_wasm.wasm')\n",
+    "wfh = wasm.WasmFileHandler(wasm_file)\n",
+    "\n",
+    "print(repr(wfh))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "f9cecb07",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def RUS(limit, cond_execute):\n",
+    "    \"\"\"\n",
+    "    limit: number of attempts\n",
+    "    cond_execute: condition to execute, i.e. run this until a total of cond_execute |1> measurements is achieved.\n",
```

Comment:
I think this RUS circuit building could use some more explanation.

Is it possible to add some explanation of what this circuit does at a basic level? The quantum part is very simple but what the classical expressions might not be clear to the average reader.

Reply:
This is a good idea, not sure if you want to take a stab at it or have Natalie do so.